experiments/tf_trainer/common/text_preprocessor.py

```diff
@@ -19,26 +34,27 @@
                          'Enable fine tuning of embeddings.')
 
 class TextPreprocessor(object):
-  """Text Preprocessor.
+  """Text Preprocessor TensorFlow Estimator Extension.
 
-  Takes an embedding and uses it to produce a word to index mapping and an
-  embedding matrix.
+  Uses embedding indexes to create tensors that map tokens (provided by an
+  abstract tokenizer funtion) to embeddings.
 
-  Note: Due to the lack of text preprocessing functions in tensorflow, we expect 
-    that the text is already preprocessed (list of words) in inference.
-    In training, due to the availability of tf.py_func, we handle the preprocessing.
+  Note: Due to the lack of text preprocessing functions in tensorflow, we expect
+  that the text is already preprocessed (list of words) in inference. In
+  training, due to the availability of tf.py_func, we can handle the
+  preprocessing.
+
+  The reasoning for this comes from this observation: The embedding step can be
```

Comment:
I am not sure to understand
Is the "word to index" the vocabulary (word--> integer)? In this case, what is the limitation around the max_words_length?